src/main/scala/cromwell/engine/WorkflowActor.scala

```diff
@@ -1,43 +1,139 @@
 
   // Note there is no Stop message defined here, Stop is from Akka.
   case object Stopped
+
+  case class Done(outputs: Map[String, WdlValue])
+
+  case class InvalidOperation(message: String)
+
+  case class Failed(throwable: Throwable)
 }
 
-// Represents the root of a single workflow instance, not a manager of multiple
-// workflows.
+/** Represents the root of a single workflow instance, not a manager of multiple
+  * workflows.
+  */
 class WorkflowActor extends Actor {
+
   import WorkflowActor._
 
+  // This var would be unnecessary in a FSM that #becomes constructed.
+  private var _constructed = false
+
+  private var _executionTable: ExecutionTable = _
+
+  private var _symbolTable: SymbolTable = _
+
+  private var _creator: ActorRef = _
+
+  /**
+   * Check inputs for presence and type compatibility.
+   * TODO break out into a designated "business object", leaving the
+   * TODO actor as a pure boundary layer object?
+   * @param workflow Workflow
+   * @param actualInputs Map of <b>local</b> input names to `WdlValue`s
+   * @return Map of locally qualified input names to a corresponding diagnostic message.
+   */
+  def checkUnsatisfiedInputs(workflow: Workflow, actualInputs: Map[FullyQualifiedName, WdlValue]): Map[FullyQualifiedName, String] =
+    workflow.inputs.collect {
+
+      case requiredInput if !actualInputs.contains(requiredInput._1) =>
+        requiredInput._1 -> "Required workflow input not specified"
+
+      case requiredInput if actualInputs.get(requiredInput._1).get.wdlType != requiredInput._2 =>
+        val expected = actualInputs.get(requiredInput._1).get.wdlType
+        // FIXME formatting
+        requiredInput._1 -> s"Incompatible workflow input types, expected $expected, got ${requiredInput._2}"
+    }
+
   override def receive: Receive = {
+    LoggingReceive {
+      case Construct(binding, actualInputs) if !_constructed =>
+        // TODO Update execution table
+        if (binding == null) {
+          sender ! InvalidOperation("Null binding")
+        } else {
+          val unsatisfiedInputs = checkUnsatisfiedInputs(binding.workflow, actualInputs)
+          if (unsatisfiedInputs.isEmpty) {
+            _constructed = true
+            _executionTable = new ExecutionTable(binding)
+            _symbolTable = new SymbolTable(binding, actualInputs)
+            _creator = sender()
+            _creator ! Constructed
+          } else {
+            sender ! ConstructionFailed(unsatisfiedInputs)
+          }
+        }
+
+      case msg if !_constructed =>
+        sender ! InvalidOperation(s"Received $msg but WorkflowActor not Constructed!")
+
+      case msg@Construct(_, _) if _constructed =>
+        sender ! InvalidOperation(s"Received $msg but WorkflowActor already Constructed!")
+
+      case Start =>
+        // TODO update execution table
+        _executionTable.runnableCalls.foreach { call =>
+          val callActor = context.actorOf(Props[CallActor], "CallActor-" + call.name)
```

Comment:
make a function in CallActor's companion object to return the props. See the warning block in the doc: http://doc.akka.io/docs/akka/current/scala/actors.html#Props